src/main/java/hudson/plugins/build_timeout/impl/DeadlineTimeOutStrategy.java

```diff
@@ -26,11 +26,11 @@
 
     public static final int MINIMUM_DEADLINE_TOLERANCE_IN_MINUTES = 1;
 
-    protected static final String DEADLINE_REGEXP = new String("[0-2]?[0-9]:[0-5][0-9](:[0-5][0-9])?");
+    protected static final String DEADLINE_REGEXP = "[0-2]?[0-9]:[0-5][0-9](:[0-5][0-9])?";
```

Comment:
DeadlineTimeOutStrategy.java:29, DM_STRING_CTOR, Priority: Normal

hudson.plugins.build_timeout.impl.DeadlineTimeOutStrategy.() invokes inefficient new String(String) constructor

Using the java.lang.String(String) constructor wastes memory because the object so constructed will be functionally indistinguishable from the String passed as a parameter.  Just use the argument String directly.